test/topology_custom/test_commitlog_segment_data_resurrection.py

```diff
@@ -0,0 +1,124 @@
+from test.pylib.manager_client import ManagerClient
+
+import pytest
+import os
+import logging
+import glob
+import json
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_pinned_cl_segment_doesnt_resurrect_data(manager: ManagerClient):
+    """
+        The tested scenario is as follows:
+        * Two tables, ks1.tbl1 and ks2.tbl2.
+        * A commitlog segment contains writes for both tables.
+        * ks1.tbl1 has very low write traffix, memtables are very rarely
+          flushed, this results in ks1.tbl1 "pinning" any commitlog segment
+          which has writes, that are still in the memtable.
+        * ks2.tbl2 deletes the data, some of which is still in the segment
+          pinned by ks1.tbl1.
+        * ks2.tbl2 flushes the memtable, data gets to sstables, commitlog
+          segments containing writes for both tables are still pinned.
+        * ks2.tbl2 compacts and purges away both the data and the tombstone.
+        * The node has an unclean restart, the pinned segments are replayed and
+          data is resurrected, as the tombstone is already purged.
+
+        This test uses gc_grace_seconds=0 to make the test fast.
```

Comment:
I think it's worth noting that gc_grace_seconds isn't just "to make the test fast" but actually in the heart of the entire scenario - whatever gc_grace_seconds is, if there's a any memtable that hasn't been flushed in the last gc_grace_seconds, we can reach this problem (so if gc_grace_seconds=0 we can reach it immediately).

Reply:
this comment has been copied from original test "as is"

Reply:
However as I see, this test would work with value gc_grace_seconds > 0 but we had to wait until grace period expired. So I guess author of this comment was right: value of 0 makes test faster